frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileUtils.java

```diff
@@ -45,4 +41,14 @@ public static List<String> getBranchNamesForPsiMacheteFile(PsiFile psiFile) {
     return List.ofAll(gitRepository.get().getInfo().getLocalBranchesWithHashes().keySet())
         .map(localBranch -> localBranch.getName());
   }
+
+  @UIThreadUnsafe
+  public static Option<GitRepository> findGitRepositoryForPsiMacheteFile(PsiFile psiFile) {
```

Comment:
Nit: we generally just return `@Nullable`s instead of `Option`s, they're typically more wieldy and &mdash; paradoxically &mdash; more null safe

Reply:
For instance, see https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1217/ + associated issues, using `Option`s can **increase** the risk of NPEs